File: rust/src/todos/todo.rs
use chrono::{DateTime, Local};
use prettytable::{row, Row};
use serde::{Deserialize, Serialize};

#[derive(Deserialize, Serialize)]
pub struct ToDo {
    number: usize,
    content: String,
    complete: bool,
    data: DateTime<Local>,
}

impl ToDo {
    pub fn new(number: usize, content: &String) -> Self {
        Self {
            number,
            content: content.clone(),
            complete: false,
            data: chrono::Local::now(),
        }
    }

    pub fn get_number(&self) -> usize {
        self.number
    }

    pub fn get_complete(&self) -> bool {
        self.complete
    }

    pub fn complete(&mut self) {
        self.complete = true
    }

    pub fn titles() -> Row {
        row!["Number", "Content", "Complete", "Date"]
    }

    pub fn row(&self) -> Row {
        let complete = match self.complete {
            true => "\u{2714}",
            false => "\u{274C}",
        };


        row![
            self.number,
            self.content,
            complete,
            chrono_humanize::HumanTime::from(self.data)
        ]
    }
}

File: go/todos/todo/todo.go
package todo

import (
	"time"

	"github.com/jedib0t/go-pretty/v6/table"
	"github.com/mergestat/timediff"
)

type ToDo struct {
	Number   uint      `csv:"number"`
	Content  string    `csv:"content"`
	Complete bool      `csv:"complete"`
	Date     time.Time `csv:"date"`
}

func New(number uint, content string) ToDo {
	return ToDo{
		Number:   number,
		Content:  content,
		Complete: false,
		Date:     time.Now(),
	}
}

func Titles() table.Row {
	return table.Row{
		"Number",
		"Content",
		"Complete",
		"Date",
	}
}

func (todo *ToDo) Row() table.Row {
	var complete string

	switch todo.Complete {
	case true:
		complete = "\u2714"
	case false:
		complete = "\u274C"
	}

	return table.Row{
		todo.Number,
		todo.Content,
		complete,
		timediff.TimeDiff(todo.Date),
	}
}
